graphql/src/entities/echo.py

```diff
@@ -0,0 +1,32 @@
+import dataclasses
+from enum import Enum
+
+import pydantic
+import strawberry
+
+
+@strawberry.enum  # Not strictly clean, but alternative is duplicating classes in graphql adapter.
+class WordTypeEnum(str, Enum):
+    # Whenever this class changes, update WordTypeEnumStrawberryType
+    palindrome = 'palindrome'
+    chaos = 'chaos'
+
+
+@strawberry.type  # Not strictly clean, but alternative is duplicating classes in graphql adapter.
+@dataclasses.dataclass
+class Echo(pydantic.BaseModel):
+    # IMPORTANT: Whenever properties are added/removed/modified/renamed, update constructor.
+    word: str
+    length: int
+    echo: str
+    word_type: WordTypeEnum
+
+    # Overriding constructor to give us type/name hints for Pydantic class.
+    def __init__(self, word: str, length: int, echo: str, word_type: WordTypeEnum):
```

Comment:
Pydantic does not provide a good constructor. It has no type hints, not even name hints. This makes EVERY argument a "magic string". Development is hard, since you need to constantly cross-reference to identify the correct argument name. If you change the name of the argument, you have to MANUALLY change it at each location the constructor is used.